File: frontend/packages/wms_core/lib/src/enums/batch_status.dart
import 'package:json_annotation/json_annotation.dart';

/// `inv_batch.status` (spec §9.2).
enum BatchStatus {
  @JsonValue('ACTIVE')
  active('ACTIVE'),
  @JsonValue('BLOCKED')
  blocked('BLOCKED'),
  @JsonValue('EXPIRED')
  expired('EXPIRED'),
  @JsonValue('QUARANTINE')
  quarantine('QUARANTINE');

  const BatchStatus(this.wire);

  final String wire;

  /// Only active batches take part in FEFO/FIFO allocation (spec §12.4).
  bool get isAllocatable => this == BatchStatus.active;

  static BatchStatus fromWire(String value) =>
      values.firstWhere((e) => e.wire == value);
}

File: backend/src/Modules/Inventory/Wms.Inventory.Domain/Enums/BatchStatus.cs
namespace Wms.Inventory.Domain.Enums;

/// <summary><c>inv_batch.status</c> (spec §9.2). Only <see cref="Active"/> batches are allocatable (spec §12.4).</summary>
public enum BatchStatus
{
    Active,
    Blocked,
    Expired,
    Quarantine,
}
